api/src/routes/challenge.ts

```diff
@@ -23,10 +30,142 @@ export const challengeRoutes: FastifyPluginCallbackTypebox = (
   _options,
   done
 ) => {
+  const challenges = getChallenges();
+
   // eslint-disable-next-line @typescript-eslint/unbound-method
   fastify.addHook('onRequest', fastify.csrfProtection);
   fastify.addHook('onRequest', fastify.authenticateSession);
 
+  fastify.post(
+    '/coderoad-challenge-completed',
+    {
+      schema: schemas.coderoadChallengeCompleted,
+      attachValidation: true
+    },
+    async (req, reply) => {
+      let userToken;
+
+      const { 'coderoad-user-token': encodedUserToken } = req.headers;
+      const { tutorialId } = req.body;
+
+      if (!tutorialId) {
+        void reply.code(400);
+        return {
+          type: 'error',
+          msg: `'tutorialId' not found in request body`
+        } as const;
+      }
+
+      if (!encodedUserToken) {
+        void reply.code(400);
+        return {
+          type: 'error',
+          msg: `'coderoad-user-token' not found in request headers`
```

Comment:
```suggestion
          msg: `'Coderoad-User-Token' not found in request headers`
```

While on the topic, can you create an issue to refactor the header to include a meaningful prefix like `FCC-` to avoid any clashes, as the header right now is too generic? Thanks.

Reply:
Will do

Reply:
This is set by CodeRoad and would have to be changed on their end.

Reply:
I'm saying that we don't have control over that header - CodeRoad sets it, so they would have to change it. CodeRoad hasn't had much work done on it in quite a while. The person who runs it has other obligations. I could ask, but would rather not. I would prefer to save any asks for something that's a bigger problem.